test/all.js

```diff
@@ -1,22 +1,9 @@
 'use strict';
 
-const metatests = require('metatests');
-const common = require('..');
-const metasync = require('metasync');
-const events = require('events');
+const fs = require('fs');
 
-global.api = { common, events, metatests, metasync };
-
-api.metatests.namespace({ common: api.common });
-
-const all = [
-  'array', 'cache', 'callback', 'curry', 'curryN',
-  'data', 'either', 'enum', 'events', 'id', 'math', 'mixin',
-  'mp', 'network', 'omap', 'partial', 'replicate', 'restLeft',
-  'safe', 'sort', 'strings', 'time', 'units', 'zip', 'zipWith',
-  'iterator', 'utilities'
-];
-
-all.forEach(name => require('./' + name));
-
-api.metatests.report();
+fs.readdir(__dirname, (err, files) => {
+  if (err) return console.error('Failed to run tests: ', err);
+  files.filter(file => file !== 'all.js')
```

Comment:
I don't like the idea of hard coding the current file's name.
Can you use `__filename` here, please?

Reply:
Oh, yeah, thanks.